independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -28,56 +30,84 @@ public class InterceptorInfo extends BeanInfo implements Comparable<InterceptorI
+            ClassInfo superClassInfo = getClassByName(beanDeployment.getBeanArchiveIndex(), superTypeName);
+            if (superClassInfo == null) {
+                throw new DefinitionException(
+                        "The interception super type " + superTypeName + " was not found in the index");
             }
+            processInheritedMethods(superClassInfo, beanDeployment);
+        }
 
-            DotName superTypeName = aClass.superName();
-            aClass = superTypeName == null || DotNames.OBJECT.equals(superTypeName) ? null
-                    : getClassByName(beanDeployment.getBeanArchiveIndex(), superTypeName);
+        for (MethodInfo method : aClass.methods()) {
+            if (Modifier.isStatic(method.flags())) {
+                continue;
+            }
+            if (method.hasAnnotation(DotNames.AROUND_INVOKE)) {
+                overrideSuperClassMethod(method);
+                aroundInvokes.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.AROUND_CONSTRUCT)) {
+                overrideSuperClassMethod(method);
+                aroundConstructs.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.POST_CONSTRUCT)) {
+                overrideSuperClassMethod(method);
+                postConstructs.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.PRE_DESTROY)) {
+                overrideSuperClassMethod(method);
+                preDestroys.add(validateSignature(method));
+            }
         }
+    }
 
-        this.aroundInvoke = aroundInvokes.isEmpty() ? null : aroundInvokes.get(0);
-        this.aroundConstruct = aroundConstructs.isEmpty() ? null : aroundConstructs.get(0);
-        this.postConstruct = postConstructs.isEmpty() ? null : postConstructs.get(0);
-        this.preDestroy = preDestroys.isEmpty() ? null : preDestroys.get(0);
-        if (aroundConstruct == null && aroundInvoke == null && preDestroy == null && postConstruct == null) {
-            LOGGER.warnf("%s declares no around-invoke method nor a lifecycle callback!", this);
+    private void overrideSuperClassMethod(MethodInfo method) {
+        overrideSuperClassMethod(method, aroundInvokes);
+        overrideSuperClassMethod(method, aroundConstructs);
+        overrideSuperClassMethod(method, postConstructs);
+        overrideSuperClassMethod(method, preDestroys);
+    }
+
+    private void overrideSuperClassMethod(MethodInfo method, List<MethodInfo> methods) {
+        for (ListIterator<MethodInfo> iterator = methods.listIterator(); iterator.hasNext();) {
+            MethodInfo interceptorMethod = iterator.next();
+            if (method.name().equals(interceptorMethod.name())
+                    && method.parameters().size() == 1
+                    && method.parameters().get(0).name() == interceptorMethod.parameters().get(0).name()
```

Comment:
I think that we can skip the param type check because all methods are already validated and both `InvocationContext` and `ArcInvocationContext` can be considered equal in this case...